File: c7_2_tree_traversal/L513-m.cpp
//
// Created by 李恒 on 2020/7/16.
//

//513. 找树左下角的值

#include <iostream>
#include <cassert>
#include <queue>
using namespace std ;

struct TreeNode{
    int val ;
    TreeNode* left ;
    TreeNode* right ;
    TreeNode(int x): val(x),left(NULL),right(NULL){}
};

class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            root = q.front();
            q.pop();
            if(root->right)
                q.push(root->right);
            if(root->left)
                q.push(root->left);
        }
        return root->val;
    }
};

File: python-algorithm/ds_5_tree/tree_2_traversal/L513-m.py
# -*- coding: utf-8 -*-

# 给定一个二叉树，在树的最后一行找到最左边的值。

class Solution:

    def findBottomLeftValue(self, root: TreeNode) -> int:
        res = root.val
        que = []
        que.append(root)
        while que!= []:
            sz = len(que)
            for i in range(sz):
                cur = que.pop(0)
                if i == 0:
                    res = cur.val
                if cur.left:
                    que.append(cur.left)
                if cur.right:
                    que.append(cur.right)
        return res